app/main.py

```diff
@@ -2,24 +2,58 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: callable, name: str) -> None:
```

Comment:
you use wrong callable
use: `from typing import Callable`
and fix in other places